x-pack/plugins/endpoint/server/routes/alerts.test.ts

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  IClusterClient,
+  IRouter,
+  IScopedClusterClient,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+  RouteConfig,
+} from 'kibana/server';
+import {
+  elasticsearchServiceMock,
+  httpServerMock,
+  httpServiceMock,
+  loggingServiceMock,
+} from '../../../../../src/core/server/mocks';
+import { AlertData, AlertResultList } from '../../common/types';
+import { SearchResponse } from 'elasticsearch';
+import { registerAlertRoutes } from './alerts';
+import { EndpointConfigSchema } from '../config';
+import * as data from '../test_data/all_alerts_data.json';
+import * as dataLegacy from '../test_data/all_alerts_data_legacy.json';
+
+describe('test alerts route', () => {
+  let routerMock: jest.Mocked<IRouter>;
+  let mockResponse: jest.Mocked<KibanaResponseFactory>;
+  let mockClusterClient: jest.Mocked<IClusterClient>;
+  let mockScopedClient: jest.Mocked<IScopedClusterClient>;
+  let routeHandler: RequestHandler<any, any, any>;
+  let routeConfig: RouteConfig<any, any, any, any>;
+
+  beforeEach(() => {
+    mockClusterClient = elasticsearchServiceMock.createClusterClient() as jest.Mocked<
+      IClusterClient
+    >;
+    mockScopedClient = elasticsearchServiceMock.createScopedClusterClient();
+    mockClusterClient.asScoped.mockReturnValue(mockScopedClient);
+    routerMock = httpServiceMock.createRouter();
+    mockResponse = httpServerMock.createResponseFactory();
+    registerAlertRoutes(routerMock, {
+      logFactory: loggingServiceMock.create(),
+      config: () => Promise.resolve(EndpointConfigSchema.validate({})),
+    });
+  });
+
+  it('should correctly calculate legacy alert total', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({});
+
+    const response: SearchResponse<AlertData> = (dataLegacy as unknown) as SearchResponse<
+      AlertData
+    >;
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(response));
+    [routeConfig, routeHandler] = routerMock.post.mock.calls.find(([{ path }]) =>
```

Comment:
By doing this, you are bypassing the validation that is built into the API you wrote

Reply:
https://github.com/elastic/kibana/pull/56538/files#diff-43c10726964ac0191be10a09ae259545R44 does not get tested this way

Reply:
Yeah, you're right. I'll look into a better way to do it. Unless you already know a better way to do it? :D

Reply:
Nope, just trying to write my own tests and noticed that this doesn't work.